test/e2e/flask/multichain-api/create-session.spec.ts

```diff
@@ -46,4 +61,144 @@ describe('Multichain API', function () {
       );
     });
   });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_createSession` with requested EVM scope that match the user’s enabled networks', function () {
+    it('should only select the specified EVM scopes requested by the user, and session scope contain valid accounts only', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          fixtures: new FixtureBuilder().withPopularNetworks().build(),
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          const INVALID_ACCOUNT = '0x9999999999999999999999999999999999999999';
+          const requestScopesToNetworkMap = {
+            'eip155:1': 'Ethereum Mainnet',
+            'eip155:42161': 'Arbitrum One',
+            'eip155:10': 'OP Mainnet',
+          };
+
+          const requestScopes = Object.keys(requestScopesToNetworkMap);
+          const networksToRequest = Object.values(requestScopesToNetworkMap);
+
+          await openMultichainDappAndConnectWalletWithExternallyConnectable(
+            driver,
+            extensionId,
+          );
+
+          await addRequestAccountsToCreateSession(driver, [
+            DEFAULT_FIXTURE_ACCOUNT,
+            INVALID_ACCOUNT,
+          ]);
+
+          await initCreateSessionScopes(driver, requestScopes);
+
+          // navigate to network selection screen
+          const editButtons = await driver.findElements('[data-testid="edit"]');
+          await editButtons[1].click();
+          await driver.delay(largeDelayMs);
+
+          await assertOnlyRequestedNetworksAreSelected(
+            driver,
+            networksToRequest,
+          );
```

Comment:
I think for this test lets skip this assertion - looking back at the user story I see why you would put this here... I think this story should be leaner - and focus just on verifying that the accounts added by the dapp ui to the create session request are correctly added to permissions (when they match an existing acount in the wallet) and ignored (when they do not match and existing account in the wallet)

Reply:
Actually lets point this assertion at the accounts instead of the networks. So something like:
- dapp requests account 3 (of 4 accounts currently available in the wallet)
- we assert that the account 3 is preselected in this flow
- in this connection flow we manually add account 2
- We assert that accounts 2 & 3 are in the session scopes returned by the wallet_createSession call

Reply:
I will do this in another `it()` then so this current one doesn't get bloated.

Reply:
Also, we currently have no setup in the e2e environment with 4 accounts injected, only 3 max. Can I adapt your case a bit or how should I proceed here ?

Reply:
Yeah maybe lets do one of these focused on accounts and one on networks?

Reply:
resolved in [de15bfb](https://github.com/MetaMask/metamask-extension/pull/29356/commits/de15bfb58aa908ad3b7f54092a123b53e963a270)